cirq_superstaq/ops/qudit_entangling_gates.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any, List, Optional, Tuple, Type
+
+import cirq
+import numpy as np
+import numpy.typing as npt
+from cirq.ops.common_gates import proper_repr
+
+import cirq_superstaq as css
+
+
+class BSwapPowGate(cirq.EigenGate):
+    """iSWAP-like qutrit entangling gate swapping the "02" and "11" states of two qutrits."""
+
+    @property
+    def dimension(self) -> int:
+        return 3
+
+    @property
+    def swapped_states(self) -> Tuple[Tuple[int, int], Tuple[int, int]]:
+        return (0, 2), (1, 1)
+
+    def _qid_shape_(self) -> Tuple[int, int]:
+        return self.dimension, self.dimension
+
+    def _swapped_state_indices(self) -> Tuple[int, int]:
+        return (
+            cirq.big_endian_digits_to_int(self.swapped_states[0], base=self._qid_shape_()),
+            cirq.big_endian_digits_to_int(self.swapped_states[1], base=self._qid_shape_()),
+        )
+
+    def _eigen_shifts(self) -> List[float]:
+        return [0.0, 0.5, -0.5]
+
+    def _eigen_components(self) -> List[Tuple[float, npt.NDArray[np.float_]]]:
+        idx0, idx1 = self._swapped_state_indices()
+
+        d = self.dimension**2
+        component_id = np.eye(d)
+        component_id[idx0, idx0] = component_id[idx1, idx1] = 0.0
+
+        component_p = np.zeros((d, d))
+        component_p[idx0, idx0] = component_p[idx1, idx1] = 0.5
+        component_p[idx0, idx1] = component_p[idx1, idx0] = 0.5
+
+        component_n = np.zeros((d, d))
+        component_n[idx0, idx0] = component_n[idx1, idx1] = 0.5
+        component_n[idx0, idx1] = component_n[idx1, idx0] = -0.5
+
+        return [(0.0, component_id), (0.5, component_p), (-0.5, component_n)]
```

Comment:
Speaking for myself, something like this (where components are explicitly identified as projectors and they clearly sum to identity) would help legibility.

```suggestion
        d = self.dimension**2

        projector_p = np.zeros((d, d))
        projector_p[idx0, idx0] = projector_p[idx1, idx1] = 0.5
        projector_p[idx0, idx1] = projector_p[idx1, idx0] = 0.5

        projector_n = np.zeros((d, d))
        projector_n[idx0, idx0] = projector_n[idx1, idx1] = 0.5
        projector_n[idx0, idx1] = projector_n[idx1, idx0] = -0.5

        projector_rest = np.eye(d) - projector_p - projector_n

        return [(0.0, projector_rest), (0.5, projector_p), (-0.5, projector_n)]
```

Reply:
very nice, i like it